route/v1/types.go

```diff
@@ -145,8 +146,140 @@ type RouteSpec struct {
+	Response []RouteHTTPHeader `json:"response" protobuf:"bytes,1,rep,name=response"`
+	// request is a list of HTTP request headers to set or delete.
+	// The actions either Set or Delete will be performed on the headers in sequence as defined in the list of request headers.
+	// A maximum of 20 request header actions may be configured.
+	// You can use this field to specify HTTP request headers that should be set or deleted
+	// when forwarding connections from the client to your application.
+	// Sample fetchers allowed are "req.hdr" and "ssl_c_der".
+	// Converters allowed are "lower" and "base64".
+	// Examples of header value - "%[req.hdr(host),lower]", "%[req.hdr(host),base64]", "{+Q}[ssl_c_der,base64]", "{+Q}[ssl_c_der,lower]"
+	// Note: This field cannot be used if your route uses TLS passthrough.
+	// + ---
+	// + Note: Any change to regex mentioned below must be reflected in the CRD validation of route in https://github.com/openshift/library-go/blob/master/pkg/route/validation/validation.go and vice-versa.
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	// +kubebuilder:validation:MaxItems=20
+	// +kubebuilder:validation:XValidation:rule=`self.all(key, key.action.type == "Delete" || (has(key.action.set) && key.action.set.value.matches('^(?:%(?:%|(?:\\{[-+]?[QXE](?:,[-+]?[QXE])*\\})?\\[(?:req\\.hdr\\([0-9A-Za-z-]+\\)|ssl_c_der)(?:,(?:lower|base64))*\\])|[^%[:cntrl:]])+$')))`,message="Either the header value provided is not in correct format or the sample fetcher/converter specified is not allowed. The dynamic header value will be interpreted as an HAProxy format string as defined in https://cbonte.github.io/haproxy-dconv/2.4/configuration.html#8.2.4 and may use HAProxy's %[] syntax and otherwise must be a valid HTTP header value as defined in https://datatracker.ietf.org/doc/html/rfc7230#section-3.2. Sample fetchers allowed are req.hdr, ssl_c_der. Converters allowed are lower, base64."
+	Request []RouteHTTPHeader `json:"request" protobuf:"bytes,2,rep,name=request"`
+}
+
+// RouteHTTPHeader specifies configuration for setting or deleting an HTTP header.
+type RouteHTTPHeader struct {
+	// name specifies a header name to be set or deleted. Its value must be a valid HTTP header.
+	// name as defined in RFC 2616 section 4.2.
```

Comment:
```suggestion
	// name specifies a header name to be set or deleted. Its value must be a valid HTTP header
	// name as defined in RFC 2616 section 4.2.
```

Reply:
done

Reply:
@miheer, the extra period came back, same as it did on the corresponding IngressController API field. https://github.com/openshift/api/compare/36a5696dd4e7c8455887e02193b18b11173605a0..bc2bded96862b9ea64bf193b75ec912ac57a93c6 removed it, and https://github.com/openshift/api/compare/8683ca29c48377acbd7279733e5b5b589b7aca92..65c7216c2269dcefe5649e1dd35694335fbe84a2 added it back again.